File: DSA Practice/LeetCode/189. Rotate Array/solution.java
class Solution {
    public void rotate(int[] nums, int k) {
        k = k%nums.length;
        int[] part1 = new int[k];
        int[] part2 = new int[nums.length - k];
        for (int i = 0; i < nums.length; i++){
            if (i < nums.length - k){
                part2[i] = nums[i];
            } else {
                part1[i-(nums.length - k)] = nums[i];
            }
        }

        for (int i = 0; i < nums.length; i++){
            if (i < k){
                nums[i] = part1[i];
            } else {
                nums[i] = part2[i-k];
            }
        }
    }
}

File: DSA Practice/LeetCode/189. Rotate Array/solution.kt
class Solution {
    fun rotate(nums: IntArray, k: Int): Unit {
        var kalt = k
        kalt = kalt%nums.size
        
        var part1 = nums.slice((nums.size - kalt)..(nums.size - 1))
        var part2 = nums.slice(0..(nums.size-kalt-1))

        for (i in 0..nums.size-1){
            if (i < kalt){
                nums.set(i, part1.get(i))
            } else {
                nums.set(i, part2.get(i-kalt))
            }
        }
    }
}